File: gcode/src/main/java/org/qw3rtrun/p3d/g/event/OKReceived.java
package org.qw3rtrun.p3d.g.event;

import org.qw3rtrun.p3d.core.msg.OKReceivedEvent;

public record OKReceived() implements OKReceivedEvent {

    @Override
    public String toString() {
        return "ok";
    }
}

File: gcode/src/main/kotlin/org/qw3rtrun/p3d/g/marlin/event/OKReceived.kt
package org.qw3rtrun.p3d.g.marlin.event

import org.qw3rtrun.p3d.core.msg.OKReceivedEvent

class OKReceived : OKReceivedEvent {
    override fun toString(): String = "ok"

    override fun equals(other: Any?): Boolean = other is OKReceived

    override fun hashCode(): Int = OKReceived::class.hashCode()

    companion object {
        @JvmField
        val INSTANCE: OKReceived = OKReceived()

        @JvmStatic
        fun getInstance(): OKReceived = INSTANCE
    }
}
